tests/erlang_tests/test_fun_info.erl

```diff
@@ -0,0 +1,88 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2024 Jakub Gonet <jakub.gonet@swmansion.com>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(test_fun_info).
+-export([start/0]).
+
+-define(SUCCESS, (0)).
+-define(ERROR, (1)).
+
+start() ->
+    try test_funs() of
+        ok -> ?SUCCESS
+    catch
+        _:E ->
+            erlang:display(E),
+            ?ERROR
+    end.
+
+f(_X, _Y, _Z) -> ok.
+
+test_funs() ->
+    LocalFun = fun(B) -> not B end,
+    LocalFunRef = fun f/3,
+    ExternalFunRef = fun erlang:apply/2,
+    NotExistingFunRef = fun erlang:undef/8,
```

Comment:
I suggest moving constants out of this function. Make a getter function that will be called using an external call (eg `?MODULE:getter_function()`) in order to avoid any possible future optimization.